packages/menu/src/MenuItem.ts

```diff
@@ -425,7 +430,10 @@ export class MenuItem extends LikeAnchor(Focusable) {
             this.handleSubmenuPointerenter
         );
         submenu.addEventListener('change', this.handleSubmenuChange);
+        submenu.setAttribute('id', `${this.id}-submenu`);
```

Comment:
This will cause unintended side effects if the consumer has already applied an ID to this element.

Reply:
@blunteshwar Please check for ID property before applying this attribute